fileshare/event_manager.go

```diff
@@ -66,15 +70,69 @@ func NewEventManager(
 	filesystem Filesystem,
 	defaultDownloadDir string,
 ) *EventManager {
-	return &EventManager{
+	em := &EventManager{
 		isProd:                isProd,
 		liveTransfers:         map[string]*LiveTransfer{},
 		transferSubscriptions: map[string]chan TransferProgressInfo{},
 		meshClient:            meshClient,
 		osInfo:                osInfo,
 		filesystem:            filesystem,
 		defaultDownloadDir:    defaultDownloadDir,
+		syncEvents:            make(chan []Event),
+		syncDoneCh:            make(chan struct{}),
+		asyncEvents:           make(chan []Event, 32),
 	}
+	go em.process()
+
+	return em
+}
+
+func (em *EventManager) process() {
+	fn := func(ev []Event) {
+		em.mutex.Lock()
+		for _, e := range ev {
+			em.handleEvent(e)
+		}
+		em.mutex.Unlock()
+	}
+
+	for {
+		select {
+		case e, ok := <-em.asyncEvents:
+			if !ok {
+				log.Println(internal.WarningPrefix, "asyncEvents channel closed")
+				return
+			}
+			fn(e)
+		case e, ok := <-em.syncEvents:
+			if !ok {
+				log.Println(internal.WarningPrefix, "syncEvents channel closed")
+				return
+			}
+			fn(e)
+			em.syncDoneCh <- struct{}{}
+		}
+	}
+}
+
+// sends an event to the event manager
```

Comment:
Make sure the code is aligned with go doc comments convention - they should start with the name of the symbol being documented